### ISSUE TITLE
Monitor junit tests/applications runs

### ISSUE THREAD

**Issue by vazexqi** [vazexqi/CodingSpectator#140]
If we captured the times when a user launches the tests or the application, we get a sense for when the user is confident/not confident about a change that was made, be it through an automated refactoring or manually.

**Comment by vazexqi** [comment]
I see that running JUnit tests have a corresponding [listener](http://help.eclipse.org/helios/index.jsp?topic=/org.eclipse.jdt.doc.isv/reference/extension-points/org_eclipse_jdt_junit_testRunListeners.html):

```
A test run listener that can be registered at
JUnitCore.addTestRunListener(TestRunListener) or as a contribution to the
org.eclipse.jdt.junit.testRunListeners extension point. The latter approach has
the advantage that the contributing plug-in is automatically loaded when a test
run starts. 
```

For executing a Java Application or Eclipse Application, I have not been able to find a generic listener. There is one for listening to [JDT breakpoints](http://help.eclipse.org/helios/index.jsp?topic=/org.eclipse.jdt.doc.isv/reference/extension-points/org_eclipse_jdt_junit_testRunListeners.html) but I am not sure how helpful that would be since the user might not be in debug mode all the time. There is the generic IVMRunner interface that we might be able to hook into but that would require changing the code in the `org.eclipse.jdt.launching` plug-in.

**Comment by Wanderer777** [comment]
Fixed by commit f77760ee4906485eec4b02a35ea65eacee69724c
